pyccel/codegen/codegen.py

```diff
@@ -207,52 +192,15 @@ def _collect_statements(self):
     def _set_kind(self):
         """Finds the source code kind."""
 
-
-        cls = (Header, EmptyNode, Comment, CommentBlock, Module)
-        is_module = all(isinstance(i,cls) for i in self.ast.body)
-
-
-
-        if is_module:
-            self._kind = 'module'
-        else:
-            self._kind = 'program'
-
-        #  ...
-
-        #  ...
-
-        expr = None
-
-        if self.is_module:
-            expr = Module(
-                self.name,
-                self.variables,
-                self.routines,
-                self.interfaces,
-                self.classes,
-                imports=self.imports)
-
-        elif self.is_program:
-            expr = Program(
-                self.name,
-                self.variables,
-                self.body.body,
-                imports=self.imports)
-
-        else:
-            raise NotImplementedError('TODO')
-
-
-        self._expr = expr
-
-        #  ...
+        self._is_program = self.ast.program is not None
+        # TODO: Expand this definition as we improve the cwrapper
+        self._is_module  = len(self.ast.funcs) + len(self.ast.interfaces) - bool(self.ast.init_func)>0
```

Comment:
this is confusing for me, I know that Any Python file can be referenced as a module, so I think we should remove `is_module`, and let only `is_program` for the case of `if __name__ == '__main__'`.

Reply:
I agree that any file can be referenced as a module, but our cwrapper doesn't work for empty modules.  I'm not sure if you can't wrap empty modules or if there's just an extra comma that shouldn't be there. In any case an empty module is not worth wrapping

Reply:
I've just checked, we can make an empty module. But I don't see why we should?

Reply:
I think this can help in the case of a package, with an empty `__init__` file, even I don't know yet how we can use empty init files in pyccelizing packages, but only we need empty modules to be accepted by pyccel.

Reply:
but one thing, if adding this feature can lead to a lot of extra changes, then I preferred to be in a separate PR.

Reply:
No it's fine it barely needs any changes. Fixed here: a22420a
And `is_module` removed here: 062b5aa

Reply:
Hmm, maybe not so easy. I am having problems linking with mpi

Reply:
We found the problem, it was nothing to do with mpi. It is now fixed